pubsub-mapped-api/src/main/java/com/google/pubsub/clients/consumer/KafkaConsumer.java

```diff
@@ -0,0 +1,592 @@
+  private StubCreator stubCreator;
+  private ImmutableMap<String, Subscription> subscriptions = ImmutableMap.of();
+
+  private final Deserializer<K> keyDeserializer;
+  private final Deserializer<V> valueDeserializer;
+
+  private final int maxPollRecords;
+
+  static {
+    GOOGLE_CLOUD_PROJECT = "projects/" + System.getenv("GOOGLE_CLOUD_PROJECT");
+    SUBSCRIPTION_PREFIX = GOOGLE_CLOUD_PROJECT + "/subscriptions/";
+    TOPIC_PREFIX = GOOGLE_CLOUD_PROJECT + "/topics/";
+  }
+
+  public KafkaConsumer(Map<String, Object> configs) {
+      this(new ConsumerConfig(configs), null, null);
+  }
+
+  public KafkaConsumer(Map<String, Object> configs,
+      Deserializer<K> keyDeserializer,
+      Deserializer<V> valueDeserializer) {
+      this(new ConsumerConfig(
+          ConsumerConfig.addDeserializerToConfig(configs, keyDeserializer, valueDeserializer)),
+          keyDeserializer, valueDeserializer);
+  }
+
+  public KafkaConsumer(Properties properties) {
+    this(new ConsumerConfig(properties), null, null);
+  }
+
+  public KafkaConsumer(Properties properties,
+      Deserializer<K> keyDeserializer,
+      Deserializer<V> valueDeserializer) {
+    this(new ConsumerConfig(
+            ConsumerConfig.addDeserializerToConfig(properties, keyDeserializer, valueDeserializer)),
+        keyDeserializer, valueDeserializer);
+  }
+
+
+  private KafkaConsumer(ConsumerConfig configs, Deserializer<K> keyDeserializer,
+      Deserializer<V> valueDeserializer) {
+    this(configs, keyDeserializer, valueDeserializer, new StubCreator(ChannelUtil.getInstance()));
+  }
+
+  @SuppressWarnings("unchecked")
+  KafkaConsumer(ConsumerConfig configs, Deserializer<K> keyDeserializer,
+      Deserializer<V> valueDeserializer, StubCreator stubCreator) {
+    try {
+      log.debug("Starting PubSub subscriber");
+
+      this.stubCreator = stubCreator;
+      this.keyDeserializer = handleDeserializer(configs,
+          ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, keyDeserializer, true);
+      this.valueDeserializer = handleDeserializer(configs,
+          ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, valueDeserializer, false);
+
+      //this is a limit on each poll for each topic
+      this.maxPollRecords = configs.getInt(ConsumerConfig.MAX_POLL_RECORDS_CONFIG);
+
+      log.debug("PubSub subscriber created");
+    } catch (Throwable t) {
+      throw new KafkaException("Failed to construct PubSub subscriber", t);
+    }
+  }
+
+  private Deserializer handleDeserializer(ConsumerConfig configs,
+      String configString, Deserializer providedDeserializer, boolean isKey) {
+    Deserializer deserializer;
+    if(providedDeserializer == null) {
+      deserializer = configs.getConfiguredInstance(configString, Deserializer.class);
+      deserializer.configure(configs.originals(), isKey);
+    } else {
+      configs.ignore(configString);
+      deserializer = providedDeserializer;
+    }
+    return deserializer;
+  }
+
+  public Set<TopicPartition> assignment() {
```

Comment:
i cannot, as it is an implementation of an interface which requires us to include a method.